app/org/sagebionetworks/bridge/models/schedules/ScheduleCriteria.java

```diff
@@ -0,0 +1,108 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.sagebionetworks.bridge.models.Criteria;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Sets;
+
+public final class ScheduleCriteria implements Criteria {
+    private final Schedule schedule;
+    private final Integer minAppVersion;
+    private final Integer maxAppVersion;
+    private final Set<String> allOfGroups;
+    private final Set<String> noneOfGroups;
+    
+    @JsonCreator
+    private ScheduleCriteria(@JsonProperty("schedule") Schedule schedule, 
+            @JsonProperty("minAppVersion") Integer minAppVersion, 
+            @JsonProperty("maxAppVersion") Integer maxAppVersion, 
+            @JsonProperty("allOfGroups") Set<String> allOfGroups, 
+            @JsonProperty("noneOfGroups") Set<String> noneOfGroups) {
+        this.schedule = schedule;
+        this.minAppVersion = minAppVersion;
+        this.maxAppVersion = maxAppVersion;
+        this.allOfGroups = (allOfGroups != null) ? allOfGroups : Sets.newHashSet();
+        this.noneOfGroups = (noneOfGroups != null) ? noneOfGroups : Sets.newHashSet();
```

Comment:
Unit test to test ScheduleCriteria with null allOf and noneOf groups and make sure they're properly converted to empty sets and pass validation?